### PR TITLE
fix: 🐛 url-encode proxy credentials

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

Proxy credentials are part of the URI put into proxy environment variables. Thus, they need to be URL encoded.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated

### CODE REVIEW

**Review by michelkaporin — APPROVED** [review]
LGTM 👍

File: src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt [hunk 1]
```diff
@@ -238,13 +238,10 @@ class ConsoleCommandRunnerTest : LightPlatformTestCase() {
...
                     assertTrue(
-                        "Should get timeout error, but received:\n$output",
-                        output.startsWith("Execution timeout")
+                        "Should get timeout error, but received:\n$output", output.startsWith("Execution timeout")
                     )
                 }
-            },
-            EmptyProgressIndicator(),
-            null
+            }, EmptyProgressIndicator(), null
```
Comment (ArtsiomCh) [inline src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt:244]: minor: a bit strange formatting here and above... does auto-formatted that way? usually arguments/params placed on separate line if not fit in one line.

**Review by ArtsiomCh — APPROVED** [review]
just minor formatting comments, otherwise looks good.

File: src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt (hunk 1 above)
Comment (bastiandoetsch) [reply src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt:244]: Yep, autoformatted according to our editorconfig.

**Comment by bastiandoetsch** [comment]
I'm waiting for a test from a customer until I merge.

File: src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt (hunk 1 above)
Comment (ArtsiomCh) [reply src/integTest/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt:244]: hmm... we might need to review our `editorconfig` as afaik that is against java/kotlin standard practices...

**Pull request merged**